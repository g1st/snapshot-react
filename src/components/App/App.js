import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Waypoint from 'react-waypoint';

import NavBar from '../NavBar';
import Hero from '../Hero';
import About from '../About';
import TeamCarousel from '../TeamCarousel';
import Gallery from '../Gallery';
import Contact from '../Contact';
import Footer from '../Footer';
import './index.css';

const propTypes = {
  onEnter: PropTypes.func,
  onLeave: PropTypes.func,
  topOffset: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

const defaultProps = { topOffset: 0 };

class App extends Component {
  state = {
    activeNav: 'home'
  };

  _handleEnterHome = () => {
    this.setState(() => ({
      activeNav: 'home'
    }));
  };

  _handleLeaveHome = () => {
    this.setState(() => ({
      activeNav: 'about'
    }));
  };

  _handleEnterAbout = () => {
    this.setState(() => ({
      activeNav: 'about'
    }));
  };

  _handleLeaveAbout = ({ currentPosition }) => {
    return currentPosition !== 'below'
      ? this.setState(() => ({
          activeNav: 'gallery'
        }))
      : null;
  };

  _handleEnterGallery = props => {
    this.setState(() => ({
      activeNav: 'gallery'
    }));
  };

  _handleLeaveGallery = ({ previousPosition }) => {
    return previousPosition === 'above'
      ? this.setState(() => ({
          activeNav: 'contact'
        }))
      : null;
  };

  _handleEnterContact = () => {
    this.setState(() => ({
      activeNav: 'contact'
    }));
  };

  render() {
    return (
      <div>
        <NavBar activeNav={this.state.activeNav} />
        <Hero />
        <Waypoint
          onEnter={this._handleEnterHome}
          onLeave={this._handleLeaveHome}
          topOffset={100}
        />
        <About />
        <TeamCarousel />
        <Waypoint
          onEnter={this._handleEnterAbout}
          onLeave={this._handleLeaveAbout}
          topOffset={100}
        />
        <Gallery />
        <Waypoint
          onEnter={this._handleEnterGallery}
          onLeave={this._handleLeaveGallery}
          topOffset={100}
        />
        <Contact />
        <Waypoint onEnter={this._handleEnterContact} topOffset={100} />
        <Footer />
      </div>
    );
  }
}

App.propTypes = propTypes;
App.defaultProps = defaultProps;

export default App;
