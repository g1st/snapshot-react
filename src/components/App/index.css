@import "~slick-carousel/slick/slick.css";
@import "~slick-carousel/slick/slick-theme.css";
* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  font-family: 'Open Sans', sans-serif;
}

.slick-dots li button:before {
  font-size: 28px;
  font-weight: bold;
  content: '‚îÅ';
}

.slick-track {
  margin-bottom: -30px;
}

:focus {
  outline: none;
}

::-moz-focus-inner {
  border: 0;
}