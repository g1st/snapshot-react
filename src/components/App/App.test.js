import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import App from '.';

describe('App Component', () => {
  it('matches snapshot', () => {
    const tree = shallow(
      <App onEnter={() => {}} onLeave={() => {}} topOffset={0} />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
