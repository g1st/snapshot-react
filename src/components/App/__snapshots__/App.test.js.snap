// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<div>
  <NavBar
    activeNav="home"
  />
  <Hero />
  <Waypoint
    bottomOffset="0px"
    fireOnRapidScroll={true}
    horizontal={false}
    onEnter={[Function]}
    onLeave={[Function]}
    onPositionChange={[Function]}
    topOffset={100}
  />
  <About />
  <TeamCarousel />
  <Waypoint
    bottomOffset="0px"
    fireOnRapidScroll={true}
    horizontal={false}
    onEnter={[Function]}
    onLeave={[Function]}
    onPositionChange={[Function]}
    topOffset={100}
  />
  <Gallery />
  <Waypoint
    bottomOffset="0px"
    fireOnRapidScroll={true}
    horizontal={false}
    onEnter={[Function]}
    onLeave={[Function]}
    onPositionChange={[Function]}
    topOffset={100}
  />
  <Contact />
  <Waypoint
    bottomOffset="0px"
    fireOnRapidScroll={true}
    horizontal={false}
    onEnter={[Function]}
    onLeave={[Function]}
    onPositionChange={[Function]}
    topOffset={100}
  />
  <Footer />
</div>
`;
